anvil-python/anvil/commands/refresh.py

```diff
@@ -0,0 +1,250 @@
+    ):
+        """Upgrade coordinator.
+
+        Execute plan for conducting an upgrade.
+
+        :client: Helper for interacting with clusterd
+        :jhelper: Helper for interacting with pylibjuju
+        :manifest: Manifest object
+        """
+        self.deployment = deployment
+        self.client = client
+        self.jhelper = jhelper
+        self.manifest = manifest
+        self.accept_defaults = accept_defaults
+        self.preseed = self.manifest.deployment_config
+
+    def run_plan(self) -> None:
+        """Execute the upgrade plan."""
+        plan = self.get_plan()
+        run_plan(plan, console)
+
+    def get_plan(self) -> list[BaseStep]:
+        """Return the upgrade plan."""
+        plan: list[BaseStep] = []
+        plan.append(LatestInChannel(self.jhelper, self.manifest))
+
+        plan.extend(
+            haproxy_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+                self.accept_defaults,
+                self.preseed,
+            )
+        )
+        plan.extend(
+            postgresql_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+                self.accept_defaults,
+                self.preseed,
+            )
+        )
+        plan.extend(
+            maas_region_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+            )
+        )
+        plan.extend(
+            maas_agent_upgrade_steps(
+                self.client,
+                self.manifest,
+                self.jhelper,
+                self.deployment.infrastructure_model,
+            )
+        )
+
+        # TODO: Update MAAS-Anvil sunbeam tag to allow using
+        # sunbeam.commands.upgrades.base.UpgradeFeatures instead of
+        # sunbeam.commands.upgrades.base.UpgradePlugins
+        # plan.append(
+        #     UpgradeFeatures(self.deployment, upgrade_release=False),
+        # )
+        plan.append(UpgradePlugins(self.deployment, upgrade_release=False))
+
+        return plan
+
+
+@click.command()
+@click.option(
+    "-m",
+    "--manifest",
+    "manifest_path",
+    help="Manifest file.",
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+)
+@click.option(
+    "-a", "--accept-defaults", help="Accept all defaults.", is_flag=True
+)
+@click.pass_context
+def refresh(
+    ctx: click.Context,
+    manifest_path: Path | None = None,
+    accept_defaults: bool = False,
+) -> None:
+    """Refresh deployment.
+
+    Refresh the deployment and allow passing new configuration options.
+    """
+
+    deployment: LocalDeployment = ctx.obj
+    client = deployment.get_client()
+
+    manifest = None
+    if manifest_path:
```

Comment:
Here we need something like the Sunbeam's refresh:

```py
if manifest:
        manifest_obj = Manifest.load(
            deployment, manifest_file=manifest, include_defaults=True
        )
        run_plan([AddManifestStep(client, manifest)], console)

if not manifest_obj:
        LOG.debug("Getting latest manifest from cluster db")
        manifest_obj = Manifest.load_latest_from_clusterdb(
            deployment, include_defaults=True
        )
```

The reason is that in case the user has not provided a manifest, we want to fetch the latest from clusterdb